xbmc/filesystem/ZipFile.cpp

```diff
@@ -305,7 +305,7 @@ ssize_t CZipFile::Read(void* lpBuf, size_t uiBufSize)
   {
     uLong iDecompressed = 0;
     uLong prevOut = m_ZStream.total_out;
-    while (((int)iDecompressed < uiBufSize) && ((m_iZipFilePos < mZipItem.csize) || (m_bFlush)))
+    while ((iDecompressed < uiBufSize) && ((m_iZipFilePos < mZipItem.csize) || (m_bFlush)))
```

Comment:
it's sign safe and I can't see anything going wrong, the PR is about warnings though and it seems like something either compiler or some static analysis tools would warn about. I don't mind if it's merged as is but I mentioned everything I spotted while going through it.